src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AddColumns.cs

```diff
@@ -86,33 +88,59 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
                 // Verify we have a string literal for the column name. Accd to spec, we don't support
                 // arbitrary expressions that evaluate to string values, because these values contribute to
                 // type analysis, so they need to be known upfront (before AddColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String ||
-                    (nameNode = nameArg.AsStrLit()) == null)
+                if (supportIndentifiers)
                 {
-                    fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
-                    continue;
+                    FirstNameNode identifierNode = nameArg.AsFirstName();
+
+                    if (identifierNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected a text literal.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
```

Comment:
>ErrExpectedStringLiteralArg_Name [](http://example.com/codeflow?start=94&length=32)

this error message is not accurate after this change, should probably be a new message capturing that it's `expecting identifier` instead.